File: backend/app/models/writing.py
from pydantic import BaseModel, Field
from typing import List, Optional
from datetime import datetime

class WritingBase(BaseModel):
    title: str
    content: str
    topic: str
    time_spent: Optional[int] = None

class WritingCreate(WritingBase):
    pass

class WritingUpdate(BaseModel):
    title: Optional[str] = None
    content: Optional[str] = None
    topic: Optional[str] = None
    time_spent: Optional[int] = None
    
    # 添加新字段用于存储AI反馈
    feedback: Optional[str] = None
    score: Optional[float] = None
    corrected_content: Optional[str] = None
    model_essay: Optional[str] = None

class Writing(WritingBase):
    id: str
    user_id: str
    created_at: datetime
    updated_at: Optional[datetime] = None
    
    # 添加新字段用于存储AI反馈
    feedback: Optional[str] = None
    score: Optional[float] = None
    corrected_content: Optional[str] = None
    model_essay: Optional[str] = None
    
    class Config:
        from_attributes = True

File: frontend/src/types/writing.ts
export interface Suggestion {
  type: 'grammar' | 'spelling' | 'style'
  text: string
  suggestion: string
  explanation: string
}

export interface WritingFeedback {
  original: string
  corrected: string
  suggestions: Suggestion[]
}

export interface WritingBase {
  title: string
  content: string
  topic?: string
  time_spent?: number
}

export interface WritingCreate extends WritingBase {}

export interface WritingUpdate {
  title?: string
  content?: string
  topic?: string
  time_spent?: number
  feedback?: string
  corrected_content?: string
  model_essay?: string
  score?: number
}

export interface Writing extends WritingBase {
  id: string
  user_id: string
  created_at: string
  feedback?: string
  corrected_content?: string
  model_essay?: string
  score?: number
}